File: escalonamento.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class EscalonadorDeProcessos {

    private static final int QUANTUM = 5; // tempo de processamento para cada processo
    private static List<Processo> listaDeProcessos = new ArrayList<>();
    private static int contadorDeProcessos = 0;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int opcao;
        do {
            System.out.println("Escolha uma opção:");
            System.out.println("1. Entrar um Processo");
            System.out.println("2. Escalonar os Processos");
            System.out.println("3. Imprimir os Processos");
            System.out.println("4. Sair");
            opcao = scanner.nextInt();

            switch (opcao) {
                case 1:
                    System.out.println("Entre com o PID:");
                    int pid = scanner.nextInt();
                    System.out.println("Entre com o Owner:");
                    String owner = scanner.next();
                    System.out.println("Entre com o Tempo do Processo:");
                    int tempoDeProcesso = scanner.nextInt();
                    Processo processo = new Processo(pid, owner, tempoDeProcesso);
                    listaDeProcessos.add(processo);
                    contadorDeProcessos++;
                    if (contadorDeProcessos % 5 == 0) {
                        System.out.println("Deseja adicionar outro processo? (S/N)");
                        String resposta = scanner.next();
                        if (resposta.equalsIgnoreCase("N")) {
                            opcao = 4; // sair
                        }
                    }
                    break;
                case 2:
                    if (listaDeProcessos.isEmpty()) {
                        System.out.println("Não há processos a serem escalonados.");
                    } else {
                        while (!listaDeProcessos.isEmpty()) {
                            for (int i = 0; i < listaDeProcessos.size(); i++) {
                                Processo processoAtual = listaDeProcessos.get(i);
                                int tempoRestante = processoAtual.getTempoDeProcesso();
                                if (tempoRestante > 0) {
                                    System.out.println("Executando processo " + processoAtual.getPid() +
                                            " (" + processoAtual.getOwner() + "), tempo restante: " + tempoRestante);
                                    tempoRestante -= QUANTUM;
                                    if (tempoRestante < 0) {
                                        tempoRestante = 0;
                                    }
                                    processoAtual.setTempoDeProcesso(tempoRestante);
                                } else {
                                    listaDeProcessos.remove(i);
                                }
                            }
                        }
                        System.out.println("Todos os processos foram executados.");
                    }
                    break;
                case 3:
                    if (listaDeProcessos.isEmpty()) {
                        System.out.println("Não há processos a serem impressos.");
                    } else {
                        System.out.println("Lista de processos:");
                        for (Processo processo : listaDeProcessos) {
                            System.out.println(processo);
                        }
                    }
                    break;
                case 4:
                    System.out.println("Saindo...");
                    break;
                default:
                    System.out.println("Opção inválida.");
            }
        } while (opcao != 4);
    }

    private static class Processo {
        private int pid;
        private String owner;
        private int tempoDeProcesso;

        public Processo(int pid, String owner, int tempoDeProcesso) {
            this.pid = pid;
            this.owner = owner;
            this.tempoDeProcesso = tempoDeProcesso;
        }

        public int getPid() {
           

File: escalonamento.py
import time

class Processo:
    def __init__(self, pid, owner, tempo, prioridade):
        self.pid = pid
        self.owner = owner
        self.tempo = tempo
        self.prioridade = prioridade

    def executar(self):
        print(f"Executando processo {self.pid} ({self.owner}), tempo restante: {self.tempo}")
        time.sleep(1)
        self.tempo -= 1

class Escalonador:
    def __init__(self):
        self.fila = []

    def adicionar_processo(self, pid, owner, tempo, prioridade):
        processo = Processo(pid, owner, tempo, prioridade)
        self.fila.append(processo)

    def imprimir_processos(self):
        print("Lista de Processos:")
        for processo in self.fila:
            print(f"PID: {processo.pid}, Owner: {processo.owner}, Tempo: {processo.tempo}, Prioridade: {processo.prioridade}")

    def escalonar_processos(self, tempo_execucao):
        while self.fila:
            processo_atual = self.fila.pop(0)
            while processo_atual.tempo > 0:
                processo_atual.executar()
                if processo_atual.prioridade > 0:
                    self.fila.insert(0, processo_atual)
                else:
                    self.fila.append(processo_atual)
                if len(self.fila) % 5 == 0:
                    resposta = input("Deseja adicionar outro processo? (s/n) ")
                    if resposta.lower() == "s":
                        self.adicionar_processo(int(input("PID: ")), input("Owner: "), int(input("Tempo: ")), int(input("Prioridade: ")))
            print(f"Processo {processo_atual.pid} ({processo_atual.owner}) finalizado.")
        print("Não existem mais processos na fila.")

escalonador = Escalonador()

while True:
    print("Selecione uma opção:")
    print("1 - Entrar um Processo")
    print("2 - Escalonar os Processos")
    print("3 - Imprimir os Processos")
    print("4 - Sair")
    opcao = int(input("Opção selecionada: "))

    if opcao == 1:
        pid = int(input("PID: "))
        owner = input("Owner: ")
        tempo = int(input("Tempo: "))
        prioridade = int(input("Prioridade: "))
        escalonador.adicionar_processo(pid, owner, tempo, prioridade)
    elif opcao == 2:
        escalonador.escalonar_processos(1)
    elif opcao == 3:
        escalonador.imprimir_processos()
    elif opcao == 4:
        break
    else:
        print("Opção inválida. Tente novamente.")
